pkg/api/handlers/libpod/pods.go

```diff
@@ -172,34 +172,19 @@ func PodStop(w http.ResponseWriter, r *http.Request) {
 			errors.Wrapf(err, "Failed to parse parameters for %s", r.URL.String()))
 		return
 	}
-	allContainersStopped := true
 	name := utils.GetName(r)
 	pod, err := runtime.LookupPod(name)
 	if err != nil {
 		utils.PodNotFound(w, name, err)
 		return
 	}
 
-	// TODO we need to implement a pod.State/Status in libpod internal so libpod api
-	// users dont have to run through all containers.
-	podContainers, err := pod.AllContainers()
+	status, err := pod.GetPodStatus()
```

Comment:
remind me, does this also account for the infra container?

Reply:
Yes, it does takes the infra container into account.  pod.GetPodStatus() and pod.AllContainers() both internally call the runtime.state.PodContainers hence no major change in the implementaion